engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -801,164 +814,43 @@ public Map<String, Object> getTableAttributes(
-        return includedRegions;
-    }
-
-    /**
-     * Get all regions and corresponding row set that overlap with the input row set. The user is responsible for
-     * closing the returned row sets.
-     *
-     * @param inputRowSet The input row set
-     * @return A list of included region information
-     */
-    private ArrayList<RegionInfoHolder> getOverlappingRegions(final RowSet inputRowSet) {
-        return getOverlappingRegions(inputRowSet, Integer.MAX_VALUE);
     }
 
     @Override
-    public void estimatePushdownFilterCost(WhereFilter filter, RowSet selection,
-            RowSet fullSet, boolean usePrev, PushdownFilterContext context, JobScheduler jobScheduler,
-            LongConsumer onComplete, Consumer<Exception> onError) {
-        final List<RegionInfoHolder> overlappingRegionsSample =
-                getOverlappingRegions(selection, PUSHDOWN_LOCATION_SAMPLES);
-        final MutableLong min = new MutableLong(Long.MAX_VALUE);
-        jobScheduler.iterateParallel(
-                ExecutionContext.getContext(),
-                logOutput -> logOutput.append("RegionedColumnSourceManager#estimatePushdownFilterCost"),
-                JobScheduler.DEFAULT_CONTEXT_FACTORY,
-                0, overlappingRegionsSample.size(),
-                (ctx, idx, locationNec, locationResume) -> {
-                    final RegionInfoHolder overlappingRegion = overlappingRegionsSample.get(idx);
-                    overlappingRegion.tle.location.estimatePushdownFilterCost(filter,
-                            overlappingRegion.rowSet, fullSet, usePrev, context, jobScheduler, value -> {
-                                synchronized (min) {
-                                    if (value < min.get()) {
-                                        min.set(value);
-                                    }
-                                }
-                                locationResume.run();
-                            }, locationNec);
-                },
-                () -> onComplete.accept(min.get()),
-                () -> SafeCloseable.closeAll(overlappingRegionsSample.iterator()),
-                (e) -> {
-                    try (final SafeCloseable ignored =
-                            () -> SafeCloseable.closeAll(overlappingRegionsSample.iterator())) {
-                        onError.accept(e);
-                    }
-                });
+    public void estimatePushdownFilterCost(
+            final WhereFilter filter,
+            final RowSet selection,
+            final boolean usePrev,
+            final PushdownFilterContext context,
+            final JobScheduler jobScheduler,
+            final LongConsumer onComplete,
+            final Consumer<Exception> onError) {
+        final int[] regionIndices = regionIndices(selection, PUSHDOWN_LOCATION_SAMPLES);
+        new EstimateJobBuilder(selection.copy(), regionIndices, onComplete, onError)
+                .iterateParallel(jobScheduler, filter, usePrev, context);
```

Comment:
The `JobBuilder` code is elegant, but we have replaced ~40 LOC with >200 LOC. I'm not certain that is a win for maintainability.